### PR TITLE
Import swedish language pack #2664

### PR DESCRIPTION
**Pull request by jswetzen**
Building and installing on my phone works, it's just the version that the app store thinks is lower than the old Swedish language pack.

It's mostly an import straight from the old repo, but I updated the Svorak layout to match Dvorak and changed the flag svg as it wouldn't generate icons properly.

### CODE REVIEW

File: addons/languages/swedish/apk/build.gradle [hunk 1]
```diff
@@ -0,0 +1,8 @@
+ext.shouldBePublished = false
```
Comment (menny) [inline addons/languages/swedish/apk/build.gradle:1]: should be `true`, we have the signing keys for this pack

File: addons/languages/swedish/apk/build.gradle [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+ext.notPublishingReason = "Initial version has not been published yet"
```
Comment (menny) [inline addons/languages/swedish/apk/build.gradle:2]: Can be removed, if we are publishing

File: addons/languages/swedish/apk/src/main/play/contact-email.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+playstore@evendanan.net
```
Comment (menny) [inline addons/languages/swedish/apk/src/main/play/contact-email.txt:1]:
```suggestion
playstore.US@evendanan.net
```

File: addons/languages/swedish/apk/src/main/play/release-notes/en-US/alpha.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+* Initial release
```
Comment (menny) [inline addons/languages/swedish/apk/src/main/play/release-notes/en-US/alpha.txt:1]:
```suggestion
* Svorak update.
```

File: addons/languages/swedish/pack/src/main/res/xml/swedish_dictionaries.xml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+ "nameResId" : the name of the dictionary. Should be short. Reference to a string in values.xml
+ "locale" : the ID of the dictionary's (ISO) locale, e.g., "en", "iw", "ar", "ru", etc. The keyboard should declare the same locale.
+ "id" : UUID for this dictionary. Obtain from here: http://www.famkruithof.net/uuid/uuidgen
+ "dictionaryResourceId" : a reference to the raw resource of the words dictionary
+ "autoTextResourceId" : a reference to a common typing mistakes XML.
+ -->
+<Dictionary nameResId="@string/dictionary_name" locale="sv" id="935fe956-c7f9-4dc6-b2d0-5a389005c826" type="binary_resource"
+		dictionaryResourceId="@array/swedish_words_dict_array" autoTextResourceId="@xml/autotext"
+		description="A swedish dictionary generated from the AOSP sources"/>
+</Dictionaries>
```
Comment (menny) [inline addons/languages/swedish/pack/src/main/res/xml/swedish_dictionaries.xml:13]: We require that all text files end with a `newline`.

File: addons/languages/swedish/pack/src/main/res/xml/swedish_keyboards.xml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<Keyboards>
+	<Keyboard nameResId="@string/swedish_keyboard" layoutResId="@xml/swedish_qwerty" id="8dd7b7fa-91c7-4be1-9b9c-cdfee1b34a47" index="1" defaultDictionaryLocale="sv" description="Created with jtlallo"/>
+	<Keyboard nameResId="@string/svorak_keyboard" layoutResId="@xml/svorak" id="595f7064-e9d8-40cc-9348-78ea065aff53" index="2" defaultDictionaryLocale="sv" description="Created by Henrik Andersson"/>
+</Keyboards>
```
Comment (menny) [inline addons/languages/swedish/pack/src/main/res/xml/swedish_keyboards.xml:5]: Same here, `newline` missing.

**Review by menny — CHANGES_REQUESTED** [review]
Great stuff!

Can you also comment here (on the PR) from where did you get the dictionary?

**Comment by jswetzen** [comment]
Thanks! The dictionary is the same as the existing Swedish pack. I downloaded from AOSP first, but it seemed identical.
-------- Originalmeddelande --------
På 13 juni 2022 19:26, Menny Even Danan skrev:

**Comment by jswetzen** [comment]
Actually when I think about it, I got curious about generating a new dictionary by combining AOSP with something like the OpenSubtitles corpus. 😏
But that's for another PR!
-------- Originalmeddelande --------
På 13 juni 2022 20:18, Menny Even Danan skrev:

**Comment by menny** [comment]
yes! go for it!
You can generate the OpenSubtitles word-list in a separate file, and our build-system will merge it with AOSP.

**Comment by menny** [comment]
But, maybe it is for another PR.

File: addons/languages/swedish/pack/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    <string name="swedish_keyboard">Swedish/Finnish</string>
+    <string name="svorak_keyboard">Svorak</string>
+    <string name="swedish_dictionary">Swedish</string>
+    <string name="store_title">Swedish Language Pack</string>
+    <string name="store_promo">Swedish language pack for AnySoftKeyboard</string>
+    <string name="store_description">AnySoftKeyboard keyboards pack:
+      Swedish keyboard and dictionary.
+
+      This is an expansion pack for AnySoftKeyboard.
+      Install AnySoftKeyboard first, and then select the desired layout from AnySoftKeyboard\'s Settings->Keyboards menu.</string>
```
[the remark this answers is not in the document]
Comment (jswetzen) [reply addons/languages/swedish/pack/src/main/res/values/strings.xml:15]: We don't! I just imported them from the existing pack and didn't think about it.

**Comment by menny** [comment]
A lot of small fixes to do here. So, I'm opening another PR: https://github.com/AnySoftKeyboard/AnySoftKeyboard/pull/3411.

I'll push the fixes on top of your commit. Thanks.

**Pull request closed without merging**